layer4/app.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 func init() {
-	caddy.RegisterModule(App{})
+	caddy.RegisterModule(&App{})
```

Comment:
As for the value/pointer receiver warnings, my approach is as follows. We do need a pointer receiver in Provision functions, so that fields could be updated in a struct. We don’t have functions where a value receiver is required, so that no field would be updated unintentionally. Golang allows structs to have both value and pointer receivers, but it’s recommended to avoid mixing them for a single struct - that’s what the warning basically says. Thus, my suggestion is to use pointer receivers everywhere unless value receivers are strictly required.

Reply:
But the module registration is never the same instance being used for provisioning. Those are separate. I'm not sure what linter you're using, but it's definitely overly aggressive on that. I don't think I agree with that rule.

Reply:
What I’m talking about is written in the golang official docs. E.g. see https://go.dev/tour/methods/8, paragraph 5. It’s a general rule to have either value or pointer receivers for **all** struct functions, but not both.

Reply:
The module is now registered here with & just because CaddyModule function has a pointer receiver instead of a value receiver it used to have. My tests show this way of module registration works fine.

Reply:
That's interesting... and yeah, I mean, it'll work... it's just unusual in terms of Caddy convention.

(I _think_ our JSON docs generator handles pointer values here correctly when it does its static analysis.)

Typically what I do is determine whether a pointer or value receiver is the correct/conservative thing to use, and choose that. If the value isn't used, or if a copy is permissible (no values from the sync package, for example, in the struct), I prefer a value receiver, I think this avoids an allocation? I'm not sure. But it also makes nil pointer derefs impossible which is nice.

Anyway... I dunno. Still feels weird to use a pointer when it's not needed.

Reply:
So we have basically 2 options here:
1) revert the "mixed value and pointer receivers" fix and keep both types of receivers for various structs of caddy-l4;
2) accept this fix and use only pointer receivers because we must have a pointer receiver in Provision functions.

If there is more support for the first option, I will adjust this PR.